web-api/src/storage.rs

```diff
@@ -143,10 +143,10 @@ impl<T> FromIterator<T> for Warning<T> {
 
 #[async_trait(?Send)]
 pub(crate) trait Document {
-    async fn get_interacted(
+    async fn get_snippets_for_interaction(
```

Comment:
Why did we call it interacted? It's not past tense? 

As we might have interaction on both snippets and documents long term I added the `snipepts` part.